docs/modules/ROOT/pages/cops_lint.adoc

```diff
@@ -3358,6 +3358,34 @@ def to_json(*_args)
 end
 ----
 
+== Lint/TopLevelReturnWithArgument
+
+|===
+| Enabled by default | Safe | Supports autocorrection | VersionAdded | VersionChanged
+
+| Pending
+| Yes
+| No
+| 0.89
+| -
+|===
+
+This cop checks for top level return with arguments. If there is a
+top-level return statement with an argument, then the argument is
+always ignored. This is detected automatically since Ruby 2.7.
+
+This cop works by validating the ancestors of the return node. A
+top-level return node's ancestors will never belong to `AST::BlockNode`
+or `AST::DefNode` class.
```

Comment:
I'd remove this comment; implementation detail is too technical and not useful to cop users.

Reply:
Makes sense to me, removed this. Added this inside the class instead.